upstream-community-operators/eclipse-che/eclipse-che.7.0.0-beta-3.0.csv.yaml

```diff
@@ -0,0 +1,358 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: cheoperator.v7.0.0-beta-3.0
+  namespace: placeholder
+  annotations:
+    alm-examples: |-
+      [
+        {
+          "apiVersion":"org.eclipse.che/v1",
+          "kind":"CheCluster",
+          "metadata":{
+             "name":"eclipse-che"
+          },
+          "spec":{
+             "server":{
+                "tlsSupport":false,
+                "selfSignedCert":false
+             },
+             "database":{
+                "externalDb":false,
+                "chePostgresHostname":"",
+                "chePostgresPort":"",
+                "chePostgresUser":"",
+                "chePostgresPassword":"",
+                "chePostgresDb":""
+             },
+             "auth":{
+                "openShiftoAuth":false,
+                "externalIdentityProvider":false,
+                "identityProviderURL":"",
+                "identityProviderRealm":"",
+                "identityProviderClientId":""
+             },
+             "storage":{
+                "pvcStrategy":"per-workspace",
+                "pvcClaimSize":"1Gi",
+                "preCreateSubPaths": true
+             }
+          }
+        }
+      ]
+    certified: "true"
+    categories: "Developer Tools, OpenShift Optional"
+    repository: https://github.com/eclipse/che-operator
+    containerImage: quay.io/eclipse/che-operator:7.0.0-beta-3.0
+    createdAt: 2019-04-06T11:59:59Z
+    description: A Kube-native development solution that delivers portable and collaborative developer workspaces in OpenShift.
+    support: Eclipse Foundation
+    capabilities: Seamless Upgrades
+spec:
+  install:
+    strategy: deployment
+    spec:
+      permissions:
+        - serviceAccountName: che-operator
+          rules:
+            - apiGroups:
+                - extensions/v1beta1
+              resources:
+                - ingresses
+              verbs:
+                - "*"
+            - apiGroups:
+                - route.openshift.io
```

Comment:
If this operator has a dependency on this openshift API, would this still work on vanilla kubernetes?

Reply:
@SamiSousa  Yes, when on k8s infra, the operator will create ingresses rather than routes. In csv, I request permissions to manage both

Reply:
When running this operator on minikube, the operator logs repeatedly print this message:
```
E0426 17:39:16.986396       1 reflector.go:134] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126: Failed to list *v1beta1.Ingress: ingresses.extensions is forbidden: User "system:serviceaccount:eclipse-che:che-operator" cannot list resource "ingresses" in API group "extensions" in the namespace "eclipse-che"
```
After deploying the example CR, it's clear to me that the operator is unable to resolve it.

Reply:
@SamiSousa thanks for checking that. Is it possible that the APIGroup is wrong? For example, in Jaeger operator it's [extensions]( https://github.com/operator-framework/community-operators/blob/master/community-operators/jaeger/jaeger.v1.11.1.clusterserviceversion.yaml#L177), while it's [extensions/v1beta1](https://github.com/operator-framework/community-operators/pull/280/files#diff-76af8518247fa4d92e16a7edd212bc28R59) for Eclipse Che. Weird thing though is that I cannot reproduce it on minikube myself, though I haven't ever tested with OLM/Marketplace on kube (only OCP/OKD).

Anyways, I have fixed API group for ingresses. I'd appreciate if you share instructions on how you test on minikube. Thanks.

Reply:
I've tested this again and still have this error:
```
ce "eclipse-che"
E0430 16:48:42.334461       1 reflector.go:134] sigs.k8s.io/controller-runtime/pkg/cache/internal/informers_map.go:126: Failed to list *v1beta1.Ingress: ingresses.extensions is forbidden: User "system:serviceaccount:eclipse-che:che-operator" cannot list resource "ingresses" in API group "extensions" in the namespace "eclipse-che"
```

Reply:
@SamiSousa are ingresses in the role that is bound to che-operator SA?

Also, can you share stes you take to test it? On MiniKube I am applying yamls with role, rolebinding and service account manually (I never tried OLM and Marketplace on k8s)

Reply:
Checked it again on MiniKube. This is the role https://github.com/eclipse/che-operator/blob/master/deploy/role.yaml, and it worked, service account can manage ingresses.